src/models/player.rs

```diff
@@ -1052,10 +1077,7 @@ fn library_item_update<E: Env + 'static>(
             meta_request: Some(meta_request),
             ..
         }) => {
-            let library_item = library_item
-                .as_ref()
-                .filter(|library_item| library_item.id == meta_request.path.id)
```

Comment:
- [ ] Before removing this we need to know why it was here in the first place and if this case still holds.

Please elaborate

Reply:
Taking a second look after discussing the flow with Tim, I think this should be fine.
THe LibraryItem form the LIbrary will be the most up-to-date library item as we send any update to be updated from the Ctx

Reply:
for the first 90 seconds of the Player lifecycle a library_item exists only in the Player, but not in ctx.library. (if you watch a movie for the first time ever)
All changes made to that temp library_item during those 90 seconds are gonna be lost. (in case youve marked video as watched)

Reply:
also `self.push_library_item_time` should be reset on Unload i think

Reply:
And   "only as temp" meaning "existing only in the player until those 90 seconds end"?

At which point it will place the library in your Library Bucket.

Reply:
yes, the library_item exits in the player for the first 90 seconds (or until you dispatch some action that triggers UpdateLibraryItem, like markvideoaswatch etc)
but with this change it recreates library_item from meta_item every time, until its not actually persisted in the library bucket